### DISCUSSION TITLE
Server Error Error: PrismaClient is unable to run in Vercel Edge Functions. As an alternative, try Accelerate: https://pris.ly/d/accelerate. If this is unexpected, please open an issue: https://github.com/prisma/prisma/issues

### DISCUSSION THREAD

**Discussion by 97revenge** [prisma/prisma#22756] [Q&A]
### Bug description

I need to validate in the database if I have the necessary id to grant access to a specific route.

### How to reproduce

```
git clone https://github.com/97revenge/auth-supabase
``` 
```
npm i 

```

```
npm run dev

```  

### Expected behavior

{
  id,
  clientGoogle,
  name,
  email,
  number,
  picture
}

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model User {
  id           Int     @id @default(autoincrement())
  clientGoogle String? @unique
  name         String?

  email   String    
  number  String?   
  picture String? 
  like    Like[]
  content Content[]

  @@map("users")
}

model Like {
  id      Int    @id @default(autoincrement())
  link    String
  owner   User   @relation(fields: [ownerId], references: [id])
  ownerId Int

  @@map("like")
}

model Content {
  id          Int    @id @default(autoincrement())
  title       String
  description String
  owner       User   @relation(fields: [ownerId], references: [id])
  ownerId     Int

  @@map("content")
}

```

```typescript
import { PrismaClient } from "@prisma/client";

import { NextResponse } from "next/server";

import type { NextRequest } from "next/server";

export default async function middleware(req: NextRequest) {
  const oauth = req.nextUrl.searchParams.get("oauth");

  const prisma = new PrismaClient();

  const instance = await prisma.user.findUnique({
    where: {
      clientGoogle: oauth as string,
    },
  });

  if (oauth) {
    return NextResponse.next();
  } else {
    return NextResponse.redirect(
      new URL(`/auth?error=${"voce precisa autenticacao para usar "}`, req.url)
    );
  }
}

export const config = {
  matcher: "/feed/:path*",
};

``` 

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

```
Environment variables loaded from .env
prisma                  : 5.8.1
@prisma/client          : 5.8.1
Computed binaryTarget   : windows
Operating System        : win32
Architecture            : x64
Node.js                 : v20.10.0
Query Engine (Node-API) : libquery-engine 78caf6feeaed953168c64e15a249c3e9a033ebe2 (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Schema Engine           : schema-engine-cli 78caf6feeaed953168c64e15a249c3e9a033ebe2 (at node_modules\@prisma\engines\schema-engine-windows.exe)
Schema Wasm             : @prisma/prisma-schema-wasm 5.8.1-1.78caf6feeaed953168c64e15a249c3e9a033ebe2
Default Engines Hash    : 78caf6feeaed953168c64e15a249c3e9a033ebe2
Studio                  : 0.497.0
```

**Comment by ludralph** [comment]
Hi @97revenge 👋 

Thank you for raising this question.

`@prisma/client` is not able to run in Vercel Edge Middleware (or Vercel Edge Functions), as mentioned in the message. The only way to currently get this working, is to use [Prisma Accelerate](https://www.prisma.io/data-platform/accelerate) and change your import to `@prisma/client/edge`. This has also been confirmed in this [issue](https://github.com/prisma/prisma/issues/20560) and [this](https://github.com/prisma/prisma/issues/21310) as well.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
